pkg/configv2/default.go

```diff
@@ -136,3 +137,29 @@ var Default = types.Bacalhau{
 		Interval: Day,
 	},
 }
+
+const defaultBacalhauDir = ".bacalhau"
+
+// getDefaultDataDir determines the appropriate default directory for storing repository data.
+// Priority order:
+// 1. If the environment variable BACALHAU_DIR is set and non-empty, use it.
+// 3. User's home directory with .bacalhau appended.
+// 4. User-specific configuration directory with .bacalhau appended.
+// 5. If all above fail, use .bacalhau in the current directory.
+// The function logs any errors encountered during the process and always returns a usable path.
+func getDefaultDataDir() string {
+	if repoDir, set := os.LookupEnv("BACALHAU_DIR"); set && repoDir != "" {
+		return repoDir
+	}
+
+	if userHome, err := os.UserHomeDir(); err == nil {
+		return filepath.Join(userHome, defaultBacalhauDir)
+	}
+
+	if userDir, err := os.UserConfigDir(); err == nil {
+		return filepath.Join(userDir, defaultBacalhauDir)
+	}
+
+	log.Error().Msg("Failed to determine default repo path. Using current directory.")
+	return defaultBacalhauDir
```

Comment:
this error log line will cause some users to panic if they already have proper config with datadir, but we just fail to determine the default repo. Lets remove this, and have a config validation at the end where we check if datadir is setup, and then can print an error message saying no dataDir was povided and we couldn't degtermin a default one

Reply:
removed the log